src/ray/gcs/gcs_server/gcs_resource_scheduler.cc

```diff
@@ -222,7 +222,7 @@ std::vector<int> GcsResourceScheduler::SortRequiredResources(
     for (auto r : extra_resources_set) {
       auto a_iter = a.custom_resources.find(r);
       const auto &a_resource = a_iter != a.custom_resources.end() ? a_iter->second : 0;
-      auto b_iter = a.custom_resources.find(r);
```

Comment:
no test caught this? If so, can we add one?

Reply:
+1. Can you add a cpp test for this case?

Reply:
Yep, will do, previously would have been caught by object store memory, but was not caught due to resource request refactoring!

Reply:
added custom resource which causes test to fail prior to this PR